gluonnlp/model/train/cache.py

```diff
@@ -0,0 +1,130 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Cache model."""
+__all__ = ['CacheCell']
+
+import mxnet as mx
+
+from mxnet import nd
+from mxnet.gluon import Block
+
+class CacheCell(Block):
+    """Cache language model.
+
+    Parameters
+    ----------
+    pretrained_lm_model : StandardRNN or AWDRNN
```

Comment:
use a name like lm_model. A block specifies the computation, so as long as the wrapped model is a language model, it should still work, regardless of whether a model is pretrained.